src/workerd/util/mem_utils.c++

```diff
@@ -0,0 +1,37 @@
+#include "mem_utils.h"
+
+#include <kj/debug.h>
+#include <kj/filesystem.h>
+
+#if !_WIN32
+#include <fcntl.h>
+#include <stdlib.h>
+#include <unistd.h>
+#endif
+
+namespace workerd::util {
+
+#if !_WIN32
+kj::Maybe<size_t> tryGetResidentSetMemory() {
+  int fd_;
+  KJ_SYSCALL(fd_ = open("/proc/self/stat", O_RDONLY));
+  kj::AutoCloseFd fd(fd_);
+  auto text = kj::FdInputStream(kj::mv(fd)).readAllText();
+  auto p = text.begin();
+  for (kj::uint i = 0; i < 23; i++) {
+    p = strchr(p, ' ');
```

Comment:
Drive-by: `kj::String` has a `find` method now, though I don't know if it's any easier to use in this case.

Reply:
Will look at that in a future iteration.